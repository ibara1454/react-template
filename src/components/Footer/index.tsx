import React from 'react';
import { NavLink } from 'react-router-dom';

interface Props {}

const Footer: React.FC<Props> = (props) => (
  <footer className="footer">
    <span className="todo-count">
      <strong>0 left</strong>
    </span>
    <ul className="filters">
      <li>
        <NavLink to="/" exact activeClassName="selected">
          All
        </NavLink>
      </li>
      <li>
        <NavLink to="/active" exact activeClassName="selected">
          Active
        </NavLink>
      </li>
      <li>
        <NavLink to="/completed" exact activeClassName="selected">
          Completed
        </NavLink>
      </li>
    </ul>
  </footer>
);

export default Footer;
