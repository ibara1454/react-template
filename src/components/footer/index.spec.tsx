import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Footer from '.';

describe('Footer', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Footer />,
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
