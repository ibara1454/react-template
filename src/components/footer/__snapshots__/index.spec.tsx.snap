// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should renders correctly 1`] = `
Array [
  <footer
    className="root"
  >
    <span
      className="count"
    >
      <strong>
        0 left
      </strong>
    </span>
    <ul
      className="filters"
    >
      <li>
        <a
          aria-current="page"
          className="link selected"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          All
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="link"
          href="/active"
          onClick={[Function]}
        >
          Active
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="link"
          href="/completed"
          onClick={[Function]}
        >
          Completed
        </a>
      </li>
    </ul>
  </footer>,
  ",",
]
`;
